### DISCUSSION TITLE
I cant seem to fix the problem: No QueryClient set, use QueryClientProvider to set one

### DISCUSSION THREAD

**Discussion by Fataydo** [wevm/wagmi#2343] [Q&A]
![image](https://github.com/wagmi-dev/wagmi/assets/73989038/a12f9a25-03b4-4327-b587-7942d4410d38)

I am using Typescript/React without next.js to Read a Contract. Even if i wrap my code around a Queryclientprovider. It doesnt seem to work and i dont understand why.
![image](https://github.com/wagmi-dev/wagmi/assets/73989038/2dfbb684-cb8e-4134-983f-412ecbc225c7)

**Comment by roblatprogen** [comment]
The callstack shows it's caused by useContractRead missing queryclient. You need QueryClientProvider to go one more level up to cover Home, so move QueryClientProvider out of Home, and wrap where Home is used instead.
